src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -235,7 +236,8 @@ private enum Dependency
                 constructedContainingTypeOfMethod,
                 formalParameterTypes,
                 formalParameterRefKinds,
-                arguments);
+                arguments,
+                binder.Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking));
```

Comment:
Note for future PR: as we discussed, I think we may be able to remove all conditional logic on the feature being enabled (both when making symbols and when inferring methods and best types).

Reply:
Added `// PROTOTYPE` comment.